### PR TITLE
Made function for time-based evolutions

### PR DESCRIPTION
**Pull request by fdeblasio**
## Description
This PR creates a function that gets the time of day and uses that function for time-based evolutions. This helps prevent needing to specify the day/night hours in multiple locations

## **Discord contact info**
Frankfurter#8828

### CODE REVIEW

**Comment by Pawkkie** [comment]
Disclaimer, I am not a reviewer! :D

I'm on board for this idea, but I'd like to put forward the suggestion of renaming the function something like IsDaytime that returns a 1 or 0 depending on the result. 

You then don't need to add more defines (it's intuitive that IsDaytime would return a 1 in the daytime), and you also don't run into function name confusion, as this function does _not_ return a time of day as its current name implies (which IS a function that a lot of day night implementations have, for example), but rather a boolean.

Great to see a new contributor!

**Comment by fdeblasio** [comment]
> Disclaimer, I am not a reviewer! :D
> 
> I'm on board for this idea, but I'd like to put forward the suggestion of renaming the function something like IsDaytime that returns a 1 or 0 depending on the result.
> 
> You then don't need to add more defines (it's intuitive that IsDaytime would return a 1 in the daytime), and you also don't run into function name confusion, as this function does _not_ return a time of day as its current name implies (which IS a function that a lot of day night implementations have, for example), but rather a boolean.
> 
> Great to see a new contributor!

Good idea! My original idea for this included dusk, but I never ended up implementing it due to the times overlapping with day/night, so with only two options, a boolean definitely works better

File: src/rtc.c [hunk 1]
```diff
@@ -293,6 +293,15 @@ void RtcCalcLocalTime(void)
...
 
+bool8 IsDaytime(void) {
+    RtcCalcLocalTime();
+    if (gLocalTime.hours >= 12 && gLocalTime.hours < 24) {
+        return TRUE;
+    } else if (gLocalTime.hours >= 0 && gLocalTime.hours < 12) {
+        return FALSE;
+    }
+}
+
```
Comment (LOuroboros) [inline src/rtc.c:304]:
```suggestion
bool8 IsDaytime(void)
{
    RtcCalcLocalTime();
    if (gLocalTime.hours >= 12 && gLocalTime.hours < 24)
        return TRUE;
    return FALSE;
}

```

**Comment by LOuroboros** [comment]
I think it may be too early to add a function like this one, personally. The expansion would get such a function in due time when a DN system is merged in. Sierra's system has one, and I'd be surprised if Merp's didn't have one too. A `GetTimeOfDay` function.
It doesn't even take into account Dusk, which is the time frame to evolve Own Tempo Rockruff.

**Comment by AsparagusEduardo** [comment]
Forgot to comment on this, sorry :(
Regardless of upcoming plans, this would've been a good thing to centralize via function ^^;
I disagree that it should've been converted to boolean, because we need to take into account the Dusk time of day.
Evolutions are based on the fact that something is night or not, so `GetTimeOfDay != TIME_NIGHT` should be enough to check for day evolutions.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Comment by Bassoonian** [comment]
You updated the times to match SwSh now, but I think a generational config that defines what time morning/day/night changes would be welcome for extra customisability.

**Comment by AsparagusEduardo** [comment]
> You updated the times to match SwSh now, but I think a generational config that defines what time morning/day/night changes would be welcome for extra customisability.

This I'm not sure. Gen 7 times were designed with the 12-hour night in mind for the inverted time (Ultra)Moon has, while previous times have the 3-hour evening that wasn't designed around Rockruff's dusk evolution.

I think SwSh strikes the balance well enough of having good night length + the evening being 1-hour long for Rockruff.

**Comment by Bassoonian** [comment]
I can see where you're coming from, but it's also a subjective thing and some people may (for whatever reason) prefer the vanilla "midnight to noon is night, everything else is day". Besides, not everyone may be planning on using Rockruff regardless. Don't we want to maximise customisability and let the users choose?

**Comment by AsparagusEduardo** [comment]
> I can see where you're coming from, but it's also a subjective thing and some people may (for whatever reason) prefer the vanilla "midnight to noon is night, everything else is day". Besides, not everyone may be planning on using Rockruff regardless. Don't we want to maximise customisability and let the users choose?

Hmm, you're right that there should be at least a vanilla setting. Alright, would be good, though not PR-blocking.

By the way @Bassoonian, can you confirm for me if the inverted (Ultra)Moon times work as intended? I'm still a bit busy a different PR ^^;

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**